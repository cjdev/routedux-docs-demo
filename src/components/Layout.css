.article-body ul {
    padding-left: 1.5em;
}

.article-body h3 {
    font-weight: bold;
}

.article-body {
    max-width: 80em;
}

.article-body h1 {
    padding-bottom: .5em;
}
