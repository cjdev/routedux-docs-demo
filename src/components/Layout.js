import React from 'react';
import { Panel, Body, Header } from '@cjdev/visual-stack/lib/components/Panel';
import ApplicationLayout from '@cjdev/visual-stack/lib/layouts/ApplicationLayout';
import "./Layout.css"

export default function Layout({content, nav, headerContent}) {

    return (
      <ApplicationLayout sideNav={nav}>
        <Panel>
            <Header>{headerContent}</Header>
            <div className="article-body">
                <Body>{content}</Body>
            </div>
        </Panel>
      </ApplicationLayout>
    );
};
