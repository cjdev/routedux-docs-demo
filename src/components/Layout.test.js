"use strict";
import React from 'react';
import ezJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import Layout from './Layout';

describe("render", () => {
   it("renders 2 slots", () => {
        //given

       //when
       const rendered = shallow(
           <Layout content={<div>Hello!</div>} nav={<ul><li>List item</li></ul>}></Layout>
       );

       //then
        expect(ezJson(rendered)).toMatchSnapshot();
   });
});
