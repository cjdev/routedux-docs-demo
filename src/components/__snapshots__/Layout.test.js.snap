// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders 2 slots 1`] = `
<ApplicationLayout
  sideNav={
    <ul>
      <li>
        List item
      </li>
    </ul>
  }
>
  <Panel>
    <Header />
    <div
      className="article-body"
    >
      <Body>
        <div>
          Hello!
        </div>
      </Body>
    </div>
  </Panel>
</ApplicationLayout>
`;
